### PR TITLE
Update publishing pages

### LINKED ISSUE #49: Update GitHub Actions OpenAPI Publishing
- [ ] Publishing to GitHub pages on merge to main
- [ ] Using latest GitHub Pages actions
- [ ] Using proper GitHub action token permissions

### PR DESCRIPTION
**Pull request by wilwade**
- Adds Dependabot
- Updates actions
- Updates github pages publishing
- Adds command to generate swagger used by the publishing service

See Test deploy: https://amplicalabs.github.io/account-service/

Closes #49

### CODE REVIEW

File: libs/common/src/config/swagger_config.ts
Comment (wilwade) [inline libs/common/src/config/swagger_config.ts:1]: Just making a function to make getting to just the doc easy

File: apps/api/src/build-openapi.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { NestFactory } from '@nestjs/core';
+
+// Mock out required env vars before the module loads
+process.env.REDIS_URL = 'http://127.0.0.1';
+process.env.FREQUENCY_URL = 'http://127.0.0.1';
+process.env.PROVIDER_ACCOUNT_SEED_PHRASE =
+  'offer debate skin describe light badge fish turtle actual inject struggle border';
+process.env.PROVIDER_ID = '0';
+process.env.PROVIDER_BASE_URL = 'http://127.0.0.1';
+process.env.CAPACITY_LIMIT = '{"type":"amount","value":0}';
```
Comment (JoeCap08055) [inline apps/api/src/build-openapi.ts:17]: This works, although another approach would be to simply commit an environment file specifically for generating the Swagger and sourcing it in the package script for the command that runs this.

**Review by JoeCap08055 — APPROVED** [review]

**Review by mattheworris — APPROVED** [review]
lgtm

File: apps/api/src/build-openapi.ts (hunk 1 above)
Comment (wilwade) [reply apps/api/src/build-openapi.ts:17]: I thought about that, but wanted it clearly connected to it, so went this path. I didn't feel strongly about it, so if someone changes it in the future, that's fine by me.

**Pull request merged**